### ISSUE TITLE
auto resize all original pictures bigger than specified size

### ISSUE THREAD

**Issue by darxmurf** [LycheeOrg/Lychee#1484]
Hi all,

It could be nice to have the possibility to force a resize of every "original" pictures to a specified size to avoid huge storage needs.

Yes it's pretty easy to resize a picture before upload but older users in the family are not always computer specialists. To keep things simple, an auto resize for every files bigger than 3000x4000 or similar could be nice.

**Comment by ildyria** [comment]
Hi, first and foremost thank you for using Lychee and for your suggestion.

I am sorry, I somewhat fail to see the usefulness of that functionality, and by that I mean how many users would benefit from it.
Furthermore I would see it as the responsibility of the user to upload reasonably size image if memory is their concern.
When exporting from e.g. Lr it is not rocket science to specify the maximum size of the picture.

For me this is way too much of a edge case. I do not know what the other think about this idea.

**Comment by darxmurf** [comment]
Thanks for your prompt answer!
Well this is clearly a "magic" functionality to help users like parent and grand-parents who are not really aware of what lightroom is or what is an image resolution and just want to strait upload from camera to website.

And in the other side, to avoid having fat images on my server.

I can potentially add a bit of code in there to adjust that myself by the way. Where is the code portion where the "full size" image is moved in it's final folder?

**Comment by ildyria** [comment]
All the full size are placed in the `upload/original/xx/xx/xxxxxxxx.jpeg`. You could write a batch script that goes through all those image, check their size and if above apply a `convert` on them.

**Comment by kamil4** [comment]
I'm more sympathetic to this suggestion, though I would phrase it a little differently:

_Why_ do we even keep the original files?

We generate up to six size variants from the original files, precisely so that we don't need to rely on them for anything. It's only if the resolution of the original images is too low to generate the intermediate sizes that we actually use the originals directly to display image previews and such. Other than that, they just sit there.

So maybe we could add a simple config that, when turned on, results in the dropping of original images at upload/import time if at least the `small` and `medium` size variants were generated successfully.

I don't think it should be the default behavior as users may want to allow the viewers to download the originals; in the future we could also need them if we add an arbitrary zoom-in capability; still, I can see how such a feature could be useful to those with limited storage space on their servers.

**Comment by darxmurf** [comment]
> So maybe we could add a simple config that, when turned on, results in the dropping of original images at upload/import time if at least the `small` and `medium` size variants were generated successfully.

This is also a pretty good option in fact!

**Comment by nagmat84** [comment]
I'm fine with that feature, too. @darxmurf If you wan't to implement it, go ahead. Unfortunately, I am too busy with other features which I consider more important. But generally, I like it.

Only one thing to note, before anybody falls into that trap: At the moment the class `SizeVariants` assumes that the original size variant is always not `null` and so does the frontend. Of course, this is not a structural problem, other size variants are nullable as well, and there is no general, technical reason why the original size variant shouldn't be nullable, too. This is just a fair warning that some more changes than a simple configuration option are required.

**Comment by kamil4** [comment]
Yes, I'm sure there are assumptions on the server side that the original size variant is non-null (e.g., rotation support), but I think the front end should be mostly fine. We do, after all, support `grants_full_photo`...
